frameworks/MLNet/exec.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import math
+import os
+import pandas as pd
+from amlb.benchmark import TaskConfig
+from amlb.data import Dataset
+from amlb.datautils import reorder_dataset
+from amlb.results import NoResultError, save_predictions
+from amlb.utils import dir_of, path_from_split, run_cmd, split_path, Timer
+import json
+from frameworks.shared.callee import call_run, result, save_metadata,  output_subdir
+
+log = logging.getLogger(__name__)
+
+def run(dataset: Dataset, config: TaskConfig):
+    log.info(f"\n**** MLNet [v{config.framework_version}] ****\n")
+    save_metadata(config)
+
+    avaible_task_list = ['classification', 'regression']
+    if config.type not in avaible_task_list:
+        raise ValueError(f'{config.type} is not supported.')
+
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    DOTNET_INSTALL_DIR = os.path.join(dir_path, 'lib')
+    os.environ['MODELBUILDER_AUTOML'] = 'NNI'
+    os.environ['DOTNET_ROOT'] = DOTNET_INSTALL_DIR
+    os.environ['MLNetCLIEnablePredict'] = 'True'
+    mlnet = os.path.join(DOTNET_INSTALL_DIR, 'mlnet')
+    temp_output_folder = output_subdir('', config=config)
+    train_time_in_seconds = config.max_runtime_seconds
+    log_path = os.path.join(temp_output_folder, 'log.txt')
+    sub_command = config.type
+    train_dataset = os.path.join(config.output_dir, config.name, f'train_{config.fold}.csv')
+    test_dataset = os.path.join(config.output_dir, config.name, f'test_{config.fold}.csv')
+    if not os.path.exists(train_dataset) or not os.path.exists(test_dataset):
+        column_num = dataset.train.X.shape[1]
+        columns=[f'column_{i}' for i in range(column_num)]
+        train_df = pd.DataFrame(dataset.train.X, columns=columns)
+        train_df['label'] = dataset.train.y
+        test_df = pd.DataFrame(dataset.test.X, columns=columns)
+        test_df['label'] = dataset.test.y
+
+        log.info(f'saving train to {train_dataset}')
+        train_df.to_csv(train_dataset, index=False, header=True)
+        log.info(f'saving test to {test_dataset}')
+        test_df.to_csv(test_dataset, index=False, header=True)
```

Comment:
a simpler alternative probably to this:
- if `dataset.train.path` extension  is `.csv` then it is already a `csv` file and the `label` column is `dataset.target.name`.
- if `dataset.train.path` extension  is `.arff` then you could use the fact that ARFF format is a superset of CSV, which would allow you to convert it to csv format without having to load it in memory: I can help providing a utility function for this if you want. We know the column names and we know that after line `@DATA` it's basically a csv file.

Reply:
I can make the change to avoid creating a new csv file from original dataset if dataset's extension is `.csv`. However, if the original dataset's extension is `.arff`, since MLNet can't load .arff dataset, we still need to save the converted dataset into a separate copy so it will still be loaded into memory. 

However, if you just want to avoid creating `Dataframe` because it's a null operation for MLNet and it's very costy for large dataset, and you want to replace it with a cheaper operation. I can follow what you suggested and it would be great if you can provide function on how to do the conversion

Reply:
> However, if you just want to avoid creating Dataframe because it's a null operation for MLNet and it's very costy for large dataset, and you want to replace it with a cheaper operation. I can follow what you suggested and it would be great if you can provide function on how to do the conversion

Yes, that's what I mean, I'll provide the function soon.